inet/vibe/inet/url.d

```diff
@@ -293,10 +293,13 @@ struct URL {
 	const {
 		import std.format;
 		auto dst = appender!string();
-		dst.put(schema);
-		dst.put(":");
-		if (isDoubleSlashSchema(schema))
-			dst.put("//");
+		if (schema.length)
+		{
+			dst.put(schema);
+			dst.put(":");
+			if (isDoubleSlashSchema(schema))
+				dst.put("//");
+		}
```

Comment:
Without the schema, the generated string is not a URL anymore. Of course it could be argued to extend this class into a full URI, but since this is just a side-effect, I'd like to keep that a separate question. Without having looked at the JS generation code too closely, to me it seems like this optimization should be handled there. If I'm not mistaken, this should also be made an opt-in/out setting, since the JS might be accessed from page hosted on a different server or sub domain.